client/pkg/schematics/schematics.go

```diff
@@ -0,0 +1,26 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package schematics contains helpers for schematic generation.
+package schematics
+
+import (
+	"strings"
+
+	"github.com/blang/semver"
+)
+
+var supportedVersion = semver.MustParse("1.7.0")
+
+// SupportsOverlays checks if Talos version supports overlays.
+func SupportsOverlays(version string) (bool, error) {
```

Comment:
fyi, we already have this in Talos: https://github.com/siderolabs/talos/blob/e23af33b59f0ac020c77dd44ce0059cb9f93e375/pkg/imager/quirks/quirks.go#L57-L65

Reply:
but I guess it's not in machinery

Reply:
oh nice. Thanks for pointing out :+1:

Reply:
But it's not in the `machinery` :(

Reply:
I can extract it though